### PR TITLE
T183130 CC payment notification w/o template

### PR DESCRIPTION
**Pull request by wiese**

### CODE REVIEW

File: src/Presentation/Presenters/CreditCardNotificationPresenter.php [hunk 1]
```diff
@@ -5,27 +5,52 @@
...
+		return $this->render( $result );
 	}
 
+	/**
+	 * Response format expected by 3rd party
+	 *
+	 * @see https://www.micropayment.de/help/documentation/
+	 */
+	private function render( array $result ): string {
+		array_walk( $result, function( & $value, $key ) {
```
Comment (gbirke) [inline src/Presentation/Presenters/CreditCardNotificationPresenter.php]: From a performance standpoint, `array_walk` with is in-place modification is probably faster than the more functional, side-effect free version:

    $values = array_map( 
          function( $value, $key) { return $key . self::VALUE_ASSIGNMENT . $value; },
          $result,
          array_keys( $result )
    );

Feel free to leave the code as it is, it's just me getting twitchy when seeing the `&` modifier in function parameters.

File: tests/EdgeToEdge/Routes/CreditCardPaymentNotificationRouteTest.php [hunk 2]
```diff
@@ -19,31 +20,32 @@
...
-				'GET',
-				'/handle-creditcard-payment-notification',
+				Request::METHOD_GET,
+				self::PATH,
 				[]
 			);
 
 			$this->assertSame( 200, $client->getResponse()->getStatusCode() );
-			$this->assertContains( 'status=error', $client->getResponse()->getContent() );
+			$this->assertContains( "status=error\n", $client->getResponse()->getContent() );
```
Comment (gbirke) [inline tests/EdgeToEdge/Routes/CreditCardPaymentNotificationRouteTest.php]: Good catch!

File: src/Presentation/Presenters/CreditCardNotificationPresenter.php (hunk 1 above)
Comment (wiese) [reply src/Presentation/Presenters/CreditCardNotificationPresenter.php]: I would have preferred a more array_map-like solution but the extra call/passing of the keys made it not worthwhile.

Comment (JeroenDeDauw) [reply src/Presentation/Presenters/CreditCardNotificationPresenter.php]: image

Comment (JeroenDeDauw) [reply src/Presentation/Presenters/CreditCardNotificationPresenter.php]: After looking at some stuff I realized array_map can take multiple arrays, so it is not that much hassle. I gave that approach a go in https://github.com/wmde/FundraisingFrontend/pull/1176. Initial result was rather verbose till I realized the reduce could be killed.

**Pull request merged**